invenio_drafts_resources/services/records/service.py

```diff
@@ -527,6 +529,16 @@ def _validate_draft(self, identity, draft):
             # explicit
         )
 
+        if draft.files.enabled:
```

Comment:
major: while in most of the cases records/drafts do contain files, that functionality is optional. For example, vocabularies are "records" that do not contain files. context: What we usually call a _record_ is a _bibliographic record_ (i.e. a publication of some sort).

Since having files is optional, the behaviour is implemented in a component. Meaning, that all the files related behaviour should be in said component. Can we move this check (I think it can be moved as is) here <https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/components/files.py#L133>?

More context: components are configured here https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/config.py#L141 and called here https://github.com/inveniosoftware/invenio-drafts-resources/blob/master/invenio_drafts_resources/services/records/service.py#L341 (you dont have to change any of this)

Reply:
Ok that makes a lot of sense, thanks a lot for the thorough/in depth explanation!
Should be all good now :)